### PR TITLE
[Polygon Edge] Add documentation on SC predeployment

### PR DESCRIPTION
**Pull request by zivkovicmilos**
# Description

This PR adds support to the Edge documentation on the newly added SC predeployment functionality.

### CODE REVIEW

File: docs/edge/additional-features/predeployment.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+
+## Overview
+
+This guide goes into detail on how to predeploy a Smart Contract on fresh Polygon Edge networks.
+
+### What is predeployment?
+
+Predeployment is a similar process to deployment - users want to have their Smart Contract present on the chain.
+Instead of manually deploying a Smart Contract on a running chain through transactions, some users want their
+Smart Contracts to be present and queriable from block `1`, at a certain address they specify.
```
Comment (ZeljkoBenovic) [inline docs/edge/additional-features/predeployment.md:23]: Typo here, should be `queryable` .

**Review by ZeljkoBenovic — CHANGES_REQUESTED** [review]
Since you're adding a new page, a `sidebars.js` entry is missing.

**Comment by Kourin1996** [comment]
I can't see new page because sidebars is not updated

File: docs/edge/additional-features/predeployment.md [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+## How do I predeploy a Smart Contract?
+
+Let's say that you have the following Smart Contract in Solidity:
+
+```solidity
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "hardhat/console.sol";
```
Comment (Kourin1996) [inline docs/edge/additional-features/predeployment.md:37]: Why import "hardhat/console.sol";?

File: docs/edge/additional-features/predeployment.md [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+  ...
+}
+```
+
+To predeploy the specified `Greeter` Smart Contract, you can run the `genesis predeploy` command:
+`polygon-edge genesis predeploy --chain ./genesis.json --predeploy-address "0x01110" --constructor-args "[[a],[b]]" --constructor-args "[123, 456]" --constructor-args "myArg"`
```
Comment (Kourin1996) [inline docs/edge/additional-features/predeployment.md:166]: Doesn't this example need `--artifacts-path`?

**Comment by laviniat1996** [comment]
@zivkovicmilos You forgot to add in sidebars.js the new page "edge/additional-features/predeployment". It won't show up otherwise.

File: docs/edge/additional-features/predeployment.md (hunk 1 above)
Comment (zivkovicmilos) [reply docs/edge/additional-features/predeployment.md]: Good catch 💯 

Fixed:

[76a9f91](https://github.com/maticnetwork/matic-docs/pull/1028/commits/76a9f91a4c6c68f218e4eeaceae00ec1bad66ee1)

File: docs/edge/additional-features/predeployment.md (hunk 2 above)
Comment (zivkovicmilos) [reply docs/edge/additional-features/predeployment.md]: It's leftover 🤦‍♂️ 

[a0556c7](https://github.com/maticnetwork/matic-docs/pull/1028/commits/a0556c7cc2b87fc702a8d0100463afbd940ed7d0)

File: docs/edge/additional-features/predeployment.md (hunk 3 above)
Comment (zivkovicmilos) [reply docs/edge/additional-features/predeployment.md]: Good catch 💯 

Added:
[2ba616d](https://github.com/maticnetwork/matic-docs/pull/1028/commits/2ba616dd94af0991a2f2ffb74de717cf3a59f3f0)

**Comment by zivkovicmilos** [comment]
@laviniat1996 @Kourin1996 

Added the missing sidebars entry 🤦‍♂️ 
[082d74a](https://github.com/maticnetwork/matic-docs/pull/1028/commits/082d74af0892367538ad8cd3fd5e970f7b9ade60)

**Review by Kourin1996 — APPROVED** [review]

**Review by ZeljkoBenovic — APPROVED** [review]
LGTM

**Pull request merged**